### ISSUE TITLE
`bar.start()` gives TypeError: write() argument must be str, not bytes

### ISSUE THREAD

**Issue by gerritholl** [WoLpH/python-progressbar#106]
#### Description

Starting with `progressbar2 3.17.0`, `bar.start()` on `Python 3.5` fails with `TypeError: write() argument must be str, not bytes`:

#### Code

```
$ cat mwe.py 
#!/usr/bin/env python

import time
import progressbar

bar = progressbar.ProgressBar(max_value=1, widgets=[progressbar.Bar("=", "[", "]"), " ", progressbar.Percentage(), ' (', progressbar.ETA(), ') '])
bar.start()
bar.update(0)
time.sleep(0.5)
bar.update(0.5)
time.sleep(0.5)
bar.update(1)
bar.finish()
$ ./mwe.py                                                                                                                                                                                                                                                                                                                                    
Traceback (most recent call last):
  File "./mwe.py", line 7, in <module>
    bar.start()
  File "/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/progressbar/bar.py", line 606, in start
    self.update(self.min_value, force=True)
  File "/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/progressbar/bar.py", line 562, in update
    StdRedirectMixin.update(self, value=value)
  File "/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/progressbar/bar.py", line 150, in update
    DefaultFdMixin.update(self, value=value)
  File "/dev/shm/gerrit/venv/stable-3.5/lib/python3.5/site-packages/progressbar/bar.py", line 55, in update
    self.fd.write(line)
TypeError: write() argument must be str, not bytes
```

#### Versions

- Python version: `3.5`
- Operating System: Red Hat Enterprise Linux Server release 6.8 (Santiago)
- Package version: `3.17.0`


The problem does not occur in `3.16.0`.

**Comment by WoLpH** [comment]
This was fixed by #105, I'll create a new release within a day but I'm currently looking for code that tests this when running the code from py.test

Your code indeed breaks for me when I do "python file.py" but it works when I run "py.test file.py"

**Comment by WoLpH** [comment]
The 3.17.1 release fixes the issue :)
